packages/java/gradle-plugin/src/main/kotlin/com/vaadin/hilla/gradle/plugin/HillaPlugin.kt

```diff
@@ -56,6 +57,16 @@ public class HillaPlugin : Plugin<Project> {
             }
         }
 
+        // Configure Kotlin-specific tasks only if Kotlin JVM plugin is applied
+        project.plugins.withId("org.jetbrains.kotlin.jvm") {
+            project.tasks.named("compileKotlin").configure { task ->
+                val compilerOptions = task.javaClass.getMethod("getCompilerOptions").invoke(task)
+                val freeCompilerArgs = compilerOptions.javaClass.getMethod("getFreeCompilerArgs")
```

Comment:
I wonder if one of those `invoke`s can return `null`

Reply:
Technically, when the Kotlin jvm plugin is applied, the compileKotlin task, and its options cannot be null. But, added null check guards for possible edge cases.